r/R/config.R

```diff
@@ -40,5 +42,7 @@ io_thread_count <- function() {
 #' @param num_threads integer: New number of threads for thread pool
 #' @export
 set_io_thread_count <- function(num_threads) {
+  current_io_thread_count <- io_thread_count()
   SetIOThreadPoolCapacity(as.integer(num_threads))
+  invisible(current_io_thread_count)
```

Comment:
Wait, are we returning the old thread count value here, or am I getting confused?

Reply:
I personally find it confusing, but I also wasn't aware of the existing behaviour so am happy to leave it given it reflects the expected behaviour.

Reply:
Yeah I actually had to spend a couple minutes parsing the tests to understand that was intended behavior. Maybe it's worth rewriting the tests to make that more obvious. For example, rewrite

```r
expect_identical(set_io_thread_count(1), current_io_thread_count)
```

to

```r
previous_count <- set_io_thread_count(1)
expect_identical(previous_count, current_io_thread_count)
```